NDTensors/src/lib/TypeParameterAccessors/test/runtests.jl

```diff
@@ -0,0 +1,180 @@
+    @test @inferred((() -> set_parameter(Vector, 2, 2))()) == Matrix
+    @test @inferred((() -> set_parameter(Array, 1, Float16))()) == Array{Float16}
+    @test @inferred((() -> set_parameter(Array, 2, 2))()) == Matrix
+
+    @test @inferred((() -> set_parameter(Vector, Position(1), Float32))()) ==
+      Vector{Float32}
+    @test @inferred ((
+      () -> set_parameter(Array{Float32}, Position(2), TypeParameter(2))
+    )()) == Array{Float32,2}
+    @test @inferred((() -> set_parameter(Array{Float32}, Float16))()) == Array{Float16}
+    @test @inferred(
+      (() -> set_parameter(Array{<:Any,2}, Position(2), TypeParameter(3)))()
+    ) == Array{<:Any,3}
+  end
+
+  @testset "Set ndim and eltype" begin
+    @test @inferred((() -> set_ndims(Array{Float32,2}, 3))()) == Array{Float32,3}
+    @test @inferred((() -> set_eltype(Array{Float32,2}, Float16))()) == Array{Float16,2}
+    @test @inferred((() -> set_ndims(Array, 2))()) == Matrix
+    @test @inferred(((() -> set_eltype(Array, Float32))())) == Array{Float32}
+    m = Transpose(Matrix{Float32}(undef, (2, 3)))
+    @test (() -> set_eltype(typeof(m), Float16))() == Transpose{Float16,Matrix{Float16}}
+    ## TODO This code does not infer the correct type but there aren't any allocations
+    ## When it is called so I believe its actually working properly
+    ## In a wrapped function on the command line this does however show the correct
+    ## value when using code_warntype.
+    @test ((() -> set_ndims(Array{<:Any,3}, 4))()) == Array{<:Any,4}
+    @test ((() -> set_eltype(Array{<:Any,3}, Float16))()) == Array{Float16,3}
+  end
+
+  @testset "Set a parameter if it is unspecified" begin
+    @test @inferred(specify_parameter(Array{Float32,3}, 1, Float16)) == Array{Float32,3}
+    @test @inferred(specify_parameter(Array{Float32,3}, 2, 2)) == Array{Float32,3}
+    @test @inferred((() -> specify_parameter(Array{Float32}, 1, Float16))()) ==
+      Array{Float32}
+    @test @inferred((() -> specify_parameter(Array{Float32}, 2, 2))()) == Array{Float32,2}
+    @test @inferred((() -> specify_parameter(Array, 1, Float16))()) == Array{Float16}
+    @test @inferred((() -> specify_parameter(Array, 2, 2))()) == Array{<:Any,2}
+    @test @inferred((() -> specify_parameter(Matrix, 1, Float32))()) == Array{Float32,2}
+    @test @inferred((() -> specify_parameter(Matrix, 2, 2))()) == Array{<:Any,2}
+    @test @inferred((() -> specify_parameter(Vector, 1, Float64))()) == Array{Float64,1}
+    @test @inferred((() -> specify_parameter(Vector, 2, 2))()) == Array{<:Any,1}
+
+    @test specify_parameter(Array{<:Any,3}, 1, Float16) == Array{Float16,3}
+    @test specify_parameter(Array{<:Any,3}, 2, 2) == Array{<:Any,3}
+    ## TODO Do we want to support type stable this behavior, it does not seem suppported by julia.
+    ## Though if you have `Matrix` or `Vector` the code is type stable
+    @test_broken @inferred(specify_parameter(Array{<:Any,3}, 1, Float16)) ==
```

Comment:
What if you use `specify_parameter(Array{<:Any,3}, Parameter(1), Float16)`?

I'm sure we can make that type stable, at the very least using a generated function but it may already be type stable since `set_parameter` is type stable for that kind of call.

Reply:
@kmp5VT is this resolved? I'm trying to close resolved or outdated comments to track what we still need to change.